### PR TITLE
Fix: Nullable introspection incorrectly squashed all inner introspection

### PR DESCRIPTION
**Pull request by nickwilliams-eventbrite**
Introspection for the new `Nullable` field was incorrectly squashing inner introspection to a Python dict string. This breaks introspection. With this change, `Nullable` introspection defers to the first positional argument, but adds a `"nullable": True` key to the returned dictionary. If the returned dictionary has no description, the `Nullable` description will be used (if set).

### CODE REVIEW

**Comment by nickwilliams-eventbrite** [comment]
>I'd rather not return it like this as a property on the other field's introspection, as it then means it's changing the schema of the other thing. Instead, let's do what things like `Any` do and embed the other introspect result inside under a key called `field` or something (or just make it pretend to be an Any to have one less introspection thing to know about)

That was my first thought, @andrewgodwin. However, I had a concern with that. Using that approach, wrapping a field with `Nullable(...)` would nest its whole introspection an entire level deeper. For displaying (as an example) auto-generated service documentation using the `introspect` action, this complicates the display further. From my perspective, this option:

```python
'my_field': {
    'type': 'dictionary',
    'contents': {
        'field1': {
            'type': 'boolean',
            'description': 'something',
        },
        'field2': {
            'type': 'unicode string',
            'description': 'Something else',
        },
    },
}
```

Seemed WAAAAAYYYYYY cleaner than this option:

```python
'my_field': {
    'type': 'nullable',
    'field': {
        'type': 'dictionary',
        'contents': {
            'field1': {
                'type': 'nullable',
                'field': {
                    'type': 'boolean',
                    'description': 'something',
                },
            },
            'field2': {
                'type': 'nullable',
                'field': {
                    'type': 'unicode string',
                    'description': 'Something else',
                },
            },
        },
    },
}
```

Thoughts based on that?

**Comment by nickwilliams-eventbrite** [comment]
Some additional comments:

In my view, introspection doesn't necessarily have to reflect the details of the implementation. Introspection is a tool to allow consumers of something (service, managed library) to determine how to use something. It also, IMO, is a foundation enabling the auto-documentation of services. As such, introspection should provide the best explanation of how something is to be used by a caller, not how nullable fields are implemented in Conformity. The first format (`'nullable': True`), I believe, more usefully conveys that information to the consumer.

tl;dr: Introspection, to me, is about the semantics of how a service or library works / is to be used, not how it's implemented.

**Comment by andrewgodwin** [comment]
In my view an introspection response has a schema, and what you've done is just add a top-level property to every single field's schema, which is especially confusing as that's not how it's modelled in Conformity (it's a wrapper, not a property).

I believe that it is much more sensible to convey consistency than to make something that's easier for one use case but actually crosses a boundary of the introspection letting you reconstruct how the schema is written.

**Comment by nickwilliams-eventbrite** [comment]
Guess I'm not gonna win this one. Updated.
